lib/matplotlib/colorbar.py

```diff
@@ -346,45 +392,132 @@ def draw_all(self):
         and do all the drawing.
         '''
 
+        # sets self._boundaries and self._values in real data units.
+        # takes into account extend values:
         self._process_values()
+        # sets self.vmin and vmax in data units, but just for
+        # the part of the colorbar that is not part of the extend
+        # patch:
         self._find_range()
+        # returns the X and Y mesh, *but* this was/is in normalized
+        # units:
         X, Y = self._mesh()
         C = self._values[:, np.newaxis]
         self._config_axes(X, Y)
         if self.filled:
             self._add_solids(X, Y, C)
+        # self._set_view_limits()
 
     def config_axis(self):
         ax = self.ax
         if self.orientation == 'vertical':
-            ax.xaxis.set_ticks([])
             # location is either one of 'bottom' or 'top'
             ax.yaxis.set_label_position(self.ticklocation)
             ax.yaxis.set_ticks_position(self.ticklocation)
+            if (isinstance(self.norm, colors.LogNorm)
+                    and self._use_adjustable()):
+                ax.set_xscale('log')
+                ax.set_yscale('log')
+            ax.xaxis.set_ticks([])
+            ax.xaxis.set_ticks([], minor=True)
+
         else:
-            ax.yaxis.set_ticks([])
             # location is either one of 'left' or 'right'
             ax.xaxis.set_label_position(self.ticklocation)
             ax.xaxis.set_ticks_position(self.ticklocation)
+            if (isinstance(self.norm, colors.LogNorm)
+                    and self._use_adjustable()):
+                ax.set_xscale('log')
+                ax.set_yscale('log')
+            ax.yaxis.set_ticks([])
+            ax.yaxis.set_ticks([], minor=True)
 
         self._set_label()
 
+    def _get_ticker_locator_formatter(self):
```

Comment:
Much of this was jut factored out of `_ticker()`  But note the new locators for LogNorm and the fallback...